public/js/eventsPage.js

```diff
@@ -92,18 +94,53 @@ function displayEventsHandler(events) {
     }
 };
 
-async function displayEvents(events) {
+async function displayEvents(events, hiringPartners) {
     events.forEach(async (event) => {
-        eventGenerator(event)
+        eventGenerator(event, hiringPartners)
+        .then((event) => {
+            let data = {
+                event_id: event.id
+            }
+
+            let hiringPartnersDiv = document.querySelector(`.hiring-partners[data-eventId='${event.id}']`)
+
+            fetch('./api/getHpsByEventId', {
```

Comment:
this can be put into a named function so its reusable and makes code less gross

Reply:
On this

Reply:
Done!